include/boost/thread/detail/config.hpp

```diff
@@ -386,6 +385,23 @@
 #endif
 
 #if defined(BOOST_THREAD_PLATFORM_WIN32)
+  #define BOOST_THREAD_HAS_MONO_CLOCK
+  #define BOOST_THREAD_INTERNAL_CLOCK_IS_MONO
+#elif defined(BOOST_THREAD_MACOS)
+  //#define BOOST_THREAD_HAS_MONO_CLOCK
+#else
+  #include <time.h> // check for CLOCK_MONOTONIC
+  #if defined(CLOCK_MONOTONIC)
+    #define BOOST_THREAD_HAS_MONO_CLOCK
+    #if defined(BOOST_THREAD_HAS_CONDATTR_SET_CLOCK_MONOTONIC)
```

Comment:
We need a way to detect this. We need a program that succeeds if everything is there and works as expected (Something like https://github.com/boostorg/thread/blob/develop/build/has_atomic_flag_lockfree_test.cpp). 

Then we can use it as in

https://github.com/boostorg/thread/blob/develop/build/Jamfile.v2#L154

and

https://github.com/boostorg/thread/blob/develop/build/Jamfile.v2#L280